README.md

```diff
@@ -173,6 +188,154 @@ Kind: *Trusted Indication*.
+
+- `recipient: u32 (Compact)`: The recipient in the to-be opened channel.
+
+Safety: The message should originate directly from the relay-chain.
+
+Kind: *System Notification*
+
+Errors:
+
+### `HrmpChannelClosing`
+
+A message to notify that the other party in an open channel decided to close it. In particular,
+`inititator` is going to close the channel opened from `sender` to the `recipient`. The close
+will be enacted at the next relay-chain session change. This message is meant to be sent by
+the relay-chain to a para.
+
+- `initiator: u32 (Compact)`: The initiator of the channel closing.
+- `sender: u32 (Compact)`: The sender in the to-be closed channel.
+- `recipient: u32 (Compact)`: The recipient in the to-be closed channel.
+
+Safety: The message should originate directly from the relay-chain.
+
+Kind: *System Notification*
+
+Errors:
+
+### `HrmpInitOpenChannel`
+
+Initiate opening a channel from a parachain to a given recipient with given channel
+parameters.
+
+- `recipient: u32 (Compact)`, - The recipient in the to-be opened channel.
+- `proposed_max_capacity: u32 (Compact)` - specifies how many messages can be in the channel at once.
+- `proposed_max_message_size: u32 (Compact)` - specifies the maximum size of any of the messages.
+
+These numbers are a subject to the relay-chain configuration limits.
+
+The channel can be opened only after the recipient confirms it and only on a session
+change.
+
+Safety: The message should originate directly from the parachain.
+
+Kind: *Instruction*.
+
+### `HrmpAcceptOpenChannel`
+
+Accept a pending open channel request from the given sender.
+
+The channel will be opened only on the next session boundary.
+
+- `sender: u32 (Compact)`: The sender in the to-be opened channel. Also, the initiator of the channel opening.
+
+Safety: The message should originate directly from the parachain.
+
+Kind: *Instruction*.
+
+Errors:
+
+### `HrmpCloseChannel`
+
+Initiate unilateral closing of a channel. The origin must be either the sender or the
+recipient in the channel being closed.
+
+The closure can only happen on a session change.
+
+- `sender: u32 (Compact)`: The sender in the to-be closed channel.
+- `recipient: u32 (Compact)`: The recipient in the to-be closed channel.
+
+Safety: The message should originate directly from the parachain.
+
+Kind: *Instruction*.
+
+Errors:
+
+### `Staking`
+
+Send staking related messages. All the messages should be executed sequentially and applied atomically.
+
+- `messages: Vec<StakingMssage>`
+
+Kind: *Meta*.
+
+Errors:
+
+#### `StakingMessage` Types
+
+##### `Bond`
+
+- `value: Compact`: The amount of token to bond.
+
+Kind: *Instruction*.
+
+##### `Unbond`
+
+- `value: Compact`: The amount of token to unbond.
+
+Kind: *Instruction*.
+
+##### `Rebond`
+
+- `value: Compact`: The amount of token to bond.
+
+Kind: *Instruction*.
+
+##### `SetPayee`
+
+- `payee: RewardDestination`: The destination account for payment.
```

Comment:
I am not very familiar with XCM: what are the assumptions? That we have two any arbitrary chains talking to each other, or two substrate based chains, or two FRAME-based chains? 

If not the former, then the design of this can be reflected on pallet-staking, and we can make sure that things that are noted here will rarely change in pallet-staking. 

Although, my guess is that this is totally generic over chain and we can't make an assumption about the existence of `pallet-staking`

Reply:
Any two arbitrary chains.